controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -518,17 +618,45 @@ void wfa_ca::handle_wfa_ca_message(
             }
         }
 
-        // TODO: Find out what to do with value of params: "backhaul", bss_info{1..N}, "name".
+        if (params.find("backhaul") != params.end()) {
+            err_string = "parameter 'backhaul' is relevant only on an agent";
+            LOG(ERROR) << err_string;
+            reply(sd, cmdu_tx, eWfaCaStatus::INVALID, err_string);
+            break;
+        }
+
+        db::bss_info_conf_t bss_info_conf;
+        std::string al_mac;
```

Comment:
These should move inside of the loop, below. Which will make it clear that in the current implementation, only one bss_info_conf gets saved in the database instead of all of them... So saving it in the database should also move inside the loop, or maybe even into the parse_bss_info function.

That does mean that if it turns out one of the parameters is invalid, we get a half-configured system. But I don't think it's important to handle that case. In practice, UCC would stop the test and eventually do a device reset, so it's not important what is in the database. And the CAPI specification doesn't say anything about what should happen in case of INVALID or ERROR.

Reply:
Yes, you are right, now that you are saying that I see that I save only the last bss_info instead of all of them. I tried to separate the input validation from the saving.
I think to change the database setter to get a list instead of one config.

Reply:
Argh, didn't find an elegant way to do it as I wanted. I'll do it as you suggested.